File: packages/sdk/sdk-ruby/tests/api_key_test.rb
require "minitest/autorun"
require_relative "../lib/index"

class ApiKeyTest < Minitest::Test
  def test_get_api_key_returns_payload
    client = AIStatsSdk::AIStats.new(
      api_key: "test",
      enable_deprecation_warnings: false
    )

    calls = []
    client.raw_client.define_singleton_method(:request) do |method:, path:, query: nil, headers: nil, body: nil|
      calls << [method, path, query, headers, body]
      {
        "data" => {
          "id" => "key_123",
          "hash" => "keyhash_123",
          "status" => "active"
        }
      }
    end

    response = client.get_api_key("key_123")

    assert_equal "key_123", response["data"]["id"]
    assert_equal "keyhash_123", response["data"]["hash"]
    assert_equal "active", response["data"]["status"]
    assert_equal [["GET", "/keys/key_123", nil, nil, nil]], calls
  end
end

File: packages/sdk/sdk-php/tests/api_key_test.php
<?php
declare(strict_types=1);

require_once __DIR__ . "/../src/index.php";

use Phaseo\Sdk\Phaseo;

function assert_true(bool $condition, string $message): void
{
    if (!$condition) {
        throw new RuntimeException($message);
    }
}

final class FakeApiKeyClient extends \Phaseo\Gen\Client
{
    public array $calls = [];

    public function __construct()
    {
        parent::__construct("https://example.test");
    }

    public function request(string $method, string $path, ?array $query = null, ?array $headers = null, $body = null)
    {
        $this->calls[] = [$method, $path, $query, $headers, $body];
        return [
            "data" => [
                "id" => "key_123",
                "hash" => "keyhash_123",
                "status" => "active",
            ],
        ];
    }
}

$client = new Phaseo(
    apiKey: "test",
    basePath: "https://api.phaseo.ai/v1",
    enableDeprecationWarnings: false
);

$fake = new FakeApiKeyClient();
$ref = new ReflectionClass($client);
$prop = $ref->getProperty("client");
$prop->setAccessible(true);
$prop->setValue($client, $fake);

$response = $client->getApiKey("key_123");
assert_true(($response["data"]["id"] ?? null) === "key_123", "expected key id");
assert_true(($response["data"]["hash"] ?? null) === "keyhash_123", "expected key hash");
assert_true(($response["data"]["status"] ?? null) === "active", "expected key status");
assert_true($fake->calls === [
    ["GET", "/keys/key_123", null, null, null],
], "expected request call for api key lookup");

echo "php api key tests ok" . PHP_EOL;
